internal/controller/ols_app_server_assets_test.go

```diff
@@ -0,0 +1,318 @@
+package controller
+
+import (
+	"path"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	olsv1alpha1 "github.com/openshift/lightspeed-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/yaml"
+)
+
+var _ = Describe("App server assets", func() {
+
+	var cr *olsv1alpha1.OLSConfig
+	var r *OLSConfigReconciler
+	var rOptions *OLSConfigReconcilerOptions
+	deploymentSelectorLabels := map[string]string{
+		"app.kubernetes.io/component":  "application-server",
+		"app.kubernetes.io/managed-by": "lightspeed-operator",
+		"app.kubernetes.io/name":       "lightspeed-service-api",
+		"app.kubernetes.io/part-of":    "openshift-lightspeed",
+	}
+
+	Context("complete custome resource", func() {
+		BeforeEach(func() {
+			rOptions = &OLSConfigReconcilerOptions{
+				LightspeedServiceImage: "lightspeed-service:latest",
+			}
+			cr = getCompleteOLSConfigCR()
+			r = &OLSConfigReconciler{
+				Options:    *rOptions,
+				logger:     logf.Log.WithName("olsconfig.reconciler"),
+				Client:     k8sClient,
```

Comment:
@raptorsun is this a fake client or real? Just trying to understand if these tests actually run on a real cluster or somewhere else.

Reply:
Hello this client is a real client from `sigs.k8s.io/controller-runtime/pkg/client`. 
The test in this file does not use the client, it is just for asset generations. 
The test in the file `ols_app_server_reconciliator_test.go` checks the assests are accepted by kube api server and created/updated correctly. The test uses [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest) that spins up an etcd server with a kubenetes api server just for the test